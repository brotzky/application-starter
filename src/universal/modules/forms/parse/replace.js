export default function(target, replacement) {
  return function(value) {
    const valueStr = value && value.toString();
    return valueStr ? valueStr.replace(target, replacement) : '';
  };
}
