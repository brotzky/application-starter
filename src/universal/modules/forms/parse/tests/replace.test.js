import replace from '../replace';

const replaceSpace = replace(/\s/, '');

describe('[Form Field Parsing] replace', () => {
  describe('replace()', () => {
    it('should return a function', () => {
      expect(replace()).toMatchSnapshot();
    });

    it('should replace only a single value that is specific', () => {
      expect(replaceSpace('1 2 3')).toMatchSnapshot();
    });

    it('should replace a value in a string that is specific', () => {
      expect(replaceSpace('1 2')).toMatchSnapshot();
    });
  });
});
