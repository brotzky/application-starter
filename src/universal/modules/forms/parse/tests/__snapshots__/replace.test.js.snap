// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Form Field Parsing] replace replace() should replace a value in a string that is specific 1`] = `"12"`;

exports[`[Form Field Parsing] replace replace() should replace only a single value that is specific 1`] = `"12 3"`;

exports[`[Form Field Parsing] replace replace() should return a function 1`] = `[Function]`;
